app/controllers/checkout_controller.rb

```diff
@@ -1,13 +1,10 @@
 class CheckoutController < ApplicationController
   def index
-    if Order.last == nil
-      redirect_to root_path
-      flash[:notice] = "You have no items in your order"
-    elsif Order.last.shopping_cart_items.count == 0
+    if Order.where(bought: false).count == 0 || Order.find_by(bought: false).shopping_cart_items.count == 0
```

Comment:
I would store the order id (of the current order you are working on) in a session and pull it up. This kind of logic makes me a bit nervous.